supervisor/resolution/evaluations/virtualization_image.py

```diff
@@ -0,0 +1,39 @@
+"""Evaluation class for virtualization image."""
+
+from ...const import CoreState
+from ...coresys import CoreSys
+from ..const import UnsupportedReason
+from .base import EvaluateBase
+
+
+def setup(coresys: CoreSys) -> EvaluateBase:
+    """Initialize evaluation-setup function."""
+    return EvaluateVirtualizationImage(coresys)
+
+
+class EvaluateVirtualizationImage(EvaluateBase):
+    """Evaluate correct OS image used when running under virtualization."""
+
+    @property
+    def reason(self) -> UnsupportedReason:
+        """Return a UnsupportedReason enum."""
+        return UnsupportedReason.VIRTUALIZATION_IMAGE
+
+    @property
+    def on_failure(self) -> str:
+        """Return a string that is printed when self.evaluate is True."""
+        return "Image of Home Assistant OS in use does not support virtualization."
+
+    @property
+    def states(self) -> list[CoreState]:
+        """Return a list of valid states when this evaluation can run."""
+        return [CoreState.STARTUP]
+
+    async def evaluate(self):
+        """Run evaluation."""
+        if not self.sys_os.available:
+            return False
+        return self.sys_host.info.virtualization and self.sys_os.board not in {
```

Comment:
I was thinking to use the Chassis property and check for `vm` here. But checking virtualization is most likely largely equivalent. It seems that if Chassis can't be read through DMI BIOS interface, it falls back to the virtualization detection (see https://github.com/systemd/systemd/blob/v252/src/hostname/hostnamed.c#L283-L286).

I guess with this we also mark container as unsupported, which most likely make sense. I don't think it is reasonable or possible to run full HAOS in a container :sweat_smile:

Reply:
So is this good then or should I change to chassis? Or check both?

Reply:
Either way should work, I think it is fine to use the virtualization property.